### PR TITLE
remove jenkins from cri-o CI

### PR DESCRIPTION
**Pull request by rphillips**
Removes jenkins from cri-o CI.

### CODE REVIEW

**Comment by haircommander** [comment]
/hold 

quick question

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml [hunk 1]
```diff
@@ -1,21 +1,5 @@
 presubmits:
   cri-o/cri-o:
-  - agent: jenkins
-    always_run: true
-    branches:
-    - ^main$
-    - ^main-
-    - ^release-1\.2$
-    - ^release-1\.2-
-    context: ci/kata-jenkins
```
Comment (haircommander) [inline ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml:10]: I think this one is running on the intel infra and we can keep it. right @mrunalp?

Comment (rphillips) [reply ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml]: fixed.

Comment (mrunalp) [reply ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml]: yeah, let's keep this one. good catch!

**Pull request merged**